momento-functions/examples/turbopuffer-index-articles.rs

```diff
@@ -0,0 +1,283 @@
+//!   -E TURBOPUFFER_ENDPOINT="$TURBOPUFFER_ENDPOINT" \
+//!   -E TURBOPUFFER_API_KEY="$TURBOPUFFER_API_KEY"
+//!
+//! # Send subset of articles for indexing via our uploaded function
+//! jq 'articles.nba' /path/to/your/data.json | curl \
+//!   curl https://api.cache.$MOMENTO_CELL_HOSTNAME/functions/$MOMENTO_CACHE_NAME/turbopuffer-index-articles \
+//!   -XPOST \
+//!   -H "authorization: $MOMENTO_API_KEY" \
+//!   -H "Content-Type: application/json" \
+//!   -d @-   
+//! ```
+
+use itertools::Itertools;
+use log::LevelFilter;
+use momento_functions::{WebResponse, WebResult};
+use momento_functions_host::{encoding::Json, web_extensions::headers};
+use momento_functions_log::LogMode;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+#[derive(Deserialize, Debug)]
+struct EmbeddingResponse {
+    data: Vec<EmbeddingData>,
+}
+#[derive(Deserialize, Serialize, Debug)]
+struct EmbeddingData {
+    embedding: Vec<f32>,
+    index: usize,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct QueryRow {
+    #[serde(alias = "$dist")]
+    dist: f32,
+    id: String,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct DocumentMetadata {
+    title: String,
+    link: String,
+    authors: Vec<String>,
+    language: String,
+    description: String,
+    feed: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct DocumentInput {
+    id: String,
+    #[serde(alias = "metadata")]
+    document_metadata: DocumentMetadata,
+    page_content: String,
+}
+
+impl DocumentInput {
+    pub fn into_turbopuffer_document(self, vector: Vec<f32>) -> TurbopufferDocument {
+        TurbopufferDocument {
+            id: self.id.clone(),
+            page_content: self.page_content.clone(),
+            vector,
+            // Flatten out the dimensions
+            metadata_title: self.document_metadata.title.clone(),
+            metadata_link: self.document_metadata.link.clone(),
+            metadata_authors: self.document_metadata.authors.clone(),
+            metadata_language: self.document_metadata.language.clone(),
+            metadata_description: self.document_metadata.description.clone(),
+            metadata_feed: self.document_metadata.feed.clone(),
+        }
+    }
+}
+
+#[derive(Serialize, Debug)]
+struct TurbopufferDocument {
+    id: String,
+    page_content: String,
+    vector: Vec<f32>,
+    // Flatten out the dimensions for easy query
+    #[serde(rename = "metadata$title")]
+    metadata_title: String,
+    #[serde(rename = "metadata$link")]
+    metadata_link: String,
+    #[serde(rename = "metadata$authors")]
+    metadata_authors: Vec<String>,
+    #[serde(rename = "metadata$language")]
+    metadata_language: String,
+    #[serde(rename = "metadata$description")]
+    metadata_description: String,
+    #[serde(rename = "metadata$feed")]
+    metadata_feed: String,
+}
+
+momento_functions::post!(index_document);
+fn index_document(Json(documents): Json<Vec<DocumentInput>>) -> WebResult<WebResponse> {
```

Comment:
Should be `index_documents`

Reply:
Oh good catch

Reply:
https://github.com/momentohq/functions/pull/4/commits/6fff10202caceff0f804eed460052e2b4e291610